File: src/commands/invite.ts
import { SlashCommandBuilder, ChatInputCommandInteraction } from "discord.js";
import { reply } from "../lib/functions/discord";

module.exports = {
	data: new SlashCommandBuilder()
		.setName("invite")
		.setDescription("Invite the bot to your server!"),

	async execute(interaction: ChatInputCommandInteraction) {
		return reply(interaction, {
			content: `https://bot.schnose.eu/`,
			ephemeral: true
		});
	}
};

File: src/commands/invite.rs
use {
	crate::{
		events::slash_commands::{InteractionState, InteractionResponseData::*},
		schnose::InteractionResult,
	},
	serenity::builder::{CreateApplicationCommand, CreateEmbed},
};

pub(crate) fn register(cmd: &mut CreateApplicationCommand) -> &mut CreateApplicationCommand {
	return cmd.name("invite").description("Invite schnose to your server!");
}

pub(crate) async fn execute(state: &mut InteractionState<'_>) -> InteractionResult {
	let embed = CreateEmbed::default()
		.color((116, 128, 194))
		.description("[click? 😳 👉👈](https://discord.com/oauth2/authorize?client_id=940308056451973120&permissions=327744&scope=bot%20applications.commands)")
		.to_owned();

	state.ephemeral();
	return Ok(Embed(embed));
}
